lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/aplinear/APLMeanSummarizer.java

```diff
@@ -70,6 +70,25 @@
         return Arrays.asList(minSupport, minStdDev);
     }
 
+    @Override
+    public double getNumberOutliers(double[][] aggregates) {
+        int n = aggregates[0].length;
+        int k = aggregates.length;
```

Comment:
Will aggregates.length ever equal 0? If so, should we initialize `k` before `n` and check? (Sorry, not super-familiar with this code.)

Reply:
In this case aggregates is created by the getAggregates function in the class so this class controls the size. It will always have 3 columns.